src/main/java/com/questhelper/requirements/item/KeyringRequirement.java

```diff
@@ -116,6 +126,16 @@ public Color getColor(Client client, QuestHelperConfig config)
 		return this.check(client) ? config.passColour() : config.failColour();
 	}
 
+	protected String getTooltipFromEnumSet(Set<TrackedContainers> containers)
+	{
+		String basicTooltip = super.getTooltipFromEnumSet(containers);
+		if (runeliteRequirement.check())
```

Comment:
Could use the `hasKeyOnKeyRing` method you just added here to make it a bit easier to read